pkg/labels/labels.go

```diff
@@ -56,11 +64,16 @@ func (ls Labels) String() string {
 		}
 		b.WriteString(l.Name)
 		b.WriteByte('=')
-		b.WriteString(strconv.Quote(l.Value))
```

Comment:
getting rid of `strconv` call here was a big perf boost

Reply:
That strconv is required for correctness, as this is used to render the labels for humans.

Reply:
is this because of what characters people can use in label values? The strconv call is very expensive in terms of allocs and cpu time. We could have a new function we don't use when rendering/logging for the strings we store in the map within promql engine?

Reply:
It's even necessary for correctness in the query engine as otherwise certain bizarre label values would be indistinguishable from regular metrics. Compare `foo="bar",dings="bums"` to `foo="bar\",dings=\"bums"`.

My suggestion is to not use the normal `String()` method for the hash replacement, but a special one that doesn't quote but uses a byte invalid in UTF-8 as a separator, similar to https://github.com/prometheus/pushgateway/blob/master/storage/diskmetricstore.go#L458-L479 .

Reply:
(Note: I'm only referring to the approach with the separator byte to avoid expensive quoting. The linked code is _not_ meant as an example for efficient string allocation handling. That's a different story.)